validator/length.go

```diff
@@ -0,0 +1,77 @@
+package validator
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/spf13/cobra"
+)
+
+type LengthRule struct {
+	Limits map[string]Limit
+}
+
+type Limit struct {
+	Min, Max int
+}
+
+func (l *LengthRule) Validate(cmd *cobra.Command) []error {
+
+	var errors []error
+	err := validateLength(cmd, l)
+	errors = append(errors, err...)
+
+	for _, child := range cmd.Commands() {
```

Comment:
we are doing only one level of commands. we should check all commands including children of children